### PR TITLE
Add custom generators for constructors

### LINKED ISSUE #14: Custom generators for certain constructors
I would like to use custom generators for some constructors of a large AST. I've tried to use `Gen1_` for the `IntegerLit` constructor [here](https://github.com/dhall-lang/dhall-haskell/commit/b0e622adc144489d329f5b6b27df4f91560900c0), but it doesn't seem to apply – the generated `IntegerLit`s are all close to 0 although the customer generator should generate only numbers close to `2^64` or close to `-(2^64)`.

Admittedly I'm not even sure whether `IntegerLit` is a "container" or not – I'm not familiar with this term.

Eventually I'd also like to use custom generators for some non-unary constructors, although I can probably get away with plugging in a custom generator for `Text`.

### PR DESCRIPTION
**Pull request by Lysxia**
Closes #14

### CODE REVIEW

**Comment by sjakobi** [comment]
Impressive! :) An example in the haddocks would be great, but I think I understand how to use this!

**Comment by Lysxia** [comment]
Just added an example!

File: src/Generic/Random/Tutorial.hs [hunk 1]
```diff
@@ -224,11 +226,56 @@
...
+--     ]
+-- @
+--
+-- So generated trees look like this ("left-biased linear trees"):
+--
+-- > Node (Node (Node (Node (Leaf 1)
+-- >                        (Leaf 2))
+-- >                  (Leaf 3))
+-- >            (Leaf 4))
+-- >      (Leaf 5)
```
Comment (sjakobi) [inline src/Generic/Random/Tutorial.hs]: I believe a right-biased tree might be easier to parse visually, but it's a great example anyway. Thanks a lot! :)

Comment (Lysxia) [reply src/Generic/Random/Tutorial.hs]: Indeed! I wanted an example where the field index was not 0 but now I'm thinking I do want a 0 there, so I'll change this example...

**Pull request merged**